tokenizers/src/models/bpe/model.rs

```diff
@@ -449,6 +462,17 @@ impl BPE {
     fn tokenize_with_cache(&self, sequence: &str) -> Result<Vec<Token>> {
         if let Some(ref hit) = self.cache.as_ref().and_then(|c| c.get(sequence)) {
             Ok(self.word_to_tokens(hit).collect())
+        } else if let Some(id) = self.vocab.get(sequence) {
+            if self.ignore_merges {
+                Ok(vec![Token::new(*id, sequence.to_string().clone(), (0, 0))])
+            } else {
+                let word = self.merge_word(sequence)?;
+                let ret = self.word_to_tokens(&word).collect();
+                if let Some(ref cache) = self.cache {
+                    cache.set(sequence.to_owned(), word);
+                }
+                Ok(ret)
```

Comment:
No don't repeat yourself :)

Reply:
Ideally check for the flag before looking into the hashmap (should be faster because of less branch mispredictions).
Not a huge deal.

Reply:
Ok

Reply:
But transfomrers is all about repeat yourself 😐